### PR TITLE
Add PII module

### PR DESCRIPTION
**Pull request by FelipeAdachi**
This PR adds a new PII module called `pii`.

It supports a much more complete and advanced pii identification than the `regexes` module, at the expense of requiring much more computational resources.

See the Notebook Example and related documentation under `modules.md` for more information.

Notes:

- Updated whylogs dependency to `^1.3.19`: Since this module creates two additional columns for each targeted column, this module uses the new `multioutput_udf`, which was introduced in whylogs 1.3.19.
- Requires extra dependencies, such as `presidio_analyzer` and `spacy`
- the `unregister_metric_udfs` function was moved from `regexes` to `utils` to make it usable to both `regexes` and `pii`.  The function was also updated to account for multioutput udfs.

### CODE REVIEW

**Review by murilommen — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -26,6 +26,8 @@ sentence-transformers = {version ="^2.2.2", optional = true}
 evaluate = {version = "^0.4.0", optional = true}
 faiss-cpu = {version = "^1.7.1", optional = true}
 ipywidgets = {version = "^8.1.1", optional = true}
+presidio-analyzer = "^2.2.351"
+h5py = "^3.10.0"
```
Comment (jamie256) [inline pyproject.toml:30]: Did you intend to make these optional? It looks like you included the dependencies below in the "all" extra which you only need to do if these are optional dependencies.

File: langkit/pii.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+def _wrapper(column):
+    def wrappee(text):
+        analyzer_results: List[tuple] = []
+        for input in text[column]:
+            analyzer_results.append(analyze_pii(input))
+        if isinstance(text, pd.DataFrame):
+            return pd.DataFrame(
+                {
+                    "result": [x[0] for x in analyzer_results],
+                    "entities_count": [x[1] for x in analyzer_results],
```
Comment (jamie256) [inline langkit/pii.py:54]: suggest making these constants since you use them more than once.

**Review by jamie256 — CHANGES_REQUESTED** [review]
Can you update the dependencies, otherwise looks good

File: pyproject.toml [hunk 3]
```diff
@@ -26,6 +26,8 @@ sentence-transformers = {version ="^2.2.2", optional = true}
 evaluate = {version = "^0.4.0", optional = true}
 faiss-cpu = {version = "^1.7.1", optional = true}
 ipywidgets = {version = "^8.1.1", optional = true}
+presidio-analyzer = "^2.2.351"
```
Comment (jamie256) [inline pyproject.toml:29]: Consider something like this, same pattern for h5py if we need that with presidio-analyzer
```suggestion
presidio-analyzer = {version = "^2.2.351", optional = true} 
```

File: pyproject.toml (hunk 1 above)
Comment (FelipeAdachi) [reply pyproject.toml]: ok, as discussed in Slack, let's keep them as optional

File: langkit/pii.py (hunk 2 above)
Comment (FelipeAdachi) [reply langkit/pii.py]: resolved

**Review by jamie256 — APPROVED** [review]
LGTM!

**Pull request merged**